ocaml/xapi/xapi_session.ml

```diff
@@ -661,6 +666,62 @@ let login_no_password_common ~__context ~uname ~originator ~host ~pool
   ignore (Client.Pool.get_all ~rpc ~session_id) ;
   session_id
 
+let login_no_password_common ~__context ~uname ~originator ~host ~pool
+    ~is_local_superuser ~subject ~auth_user_sid ~auth_user_name
+    ~rbac_permissions ~db_ref ~client_certificate =
+  Context.with_tracing ~originator ~__context __FUNCTION__ @@ fun __context ->
+  let is_useable_session session_id =
+    let is_valid_session session_id =
+      try
+        (* Call an API function to check the session is still valid *)
+        let rpc = Helpers.make_rpc ~__context in
+        ignore (Client.Pool.get_all ~rpc ~session_id) ;
```

Comment:
Is this call reasonably cheap (since the result is thrown away)?

Reply:
It's the same call that is used after creating a session to force the time to update (see line 666 above), so I assume it is the cheapest call we can do. It does take some time though, which is why I've disabled calling is_valid_session by default (!Xapi_globs.validate_reusable_pool_session defaults to false)

Reply:
One possibility where the session would become invalid is if the coordinator gets restarted (all sessions are gone). Would be worth checking that pool members can still recover from that.

Reply:
Maybe in database/master_connection.ml in the reconnect() function would need a new hook added so that when the master connection is lost and restored the global reusable session is set to null.